src/main/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImpl.java

```diff
@@ -0,0 +1,371 @@
+        filing.setDescription(getFilingDescription(accountsType, periodEndDate));
+        filing.setDescriptionValues(getDescriptionValues(periodEndDate));
+        filing.setData(getFilingData(periodEndDate, ixbrlLocation));
+
+        return filing;
+    }
+
+    /**
+     * Get the filing data, it contains period end date and the links (ixbrl location and
+     * relationship link).
+     *
+     * @param periodEndDate - accounts end date
+     * @param ixbrlLocation - the location where ixbrl is stored
+     * @return {@link Data}
+     */
+    private Data getFilingData(LocalDate periodEndDate, String ixbrlLocation) {
+        Data data = new Data();
+        data.setPeriodEndOn(periodEndDate);
+        data.setLinks(getFilingLinks(ixbrlLocation));
+
+        return data;
+    }
+
+    /**
+     * Get the Link containing the ixbrl location and the relationship link e.g. accounts.
+     *
+     * @param ixbrlLocation - the location where ixbrl is stored
+     * @return {@link List<Link>}
+     */
+    private List<Link> getFilingLinks(String ixbrlLocation) {
+        Link link = new Link();
+        link.setRelationship(LINK_RELATIONSHIP);
+        link.setHref(ixbrlLocation);
+
+        return Arrays.asList(link);
+    }
+
+    /**
+     * Get the description values, which currently contains the period end date. This data is
+     * required for the filing description.
+     *
+     * @param periodEndDate - account's period end date
+     * @return {@link Map<String,String>} containing period end date, to match filing model
+     */
+    private Map<String, String> getDescriptionValues(LocalDate periodEndDate) {
+        Map<String, String> descriptionValues = new HashMap<>();
+        descriptionValues.put(PERIOD_END_ON, periodEndDate.toString());
+
+        return descriptionValues;
+    }
+
+    /**
+     * Get the description for the filing. The description for the account type is retrieved by
+     * using DocumentDescriptionHelper class.
+     *
+     * @param accountsType - Account type information: account type, ixbrl's template name, account
+     * @param periodEndDate - account's period end date
+     * @return accounts description.
+     * @throws IOException
+     */
+    private String getFilingDescription(AccountsType accountsType, LocalDate periodEndDate)
+        throws IOException {
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put(PERIOD_END_ON, periodEndDate);
+
+        return DocumentDescriptionHelper
+            .getDescription(accountsType.getFilingDescriptionKey(), parameters);
+    }
+
+    /**
+     * Validates the ixbrl against TNEP. This validation is driven by the env. variable and it can
+     * be disable.
+     */
+    private boolean isValidIXBL() {
+
+        boolean isIxbrlValid = true;
+        if ("true".equals(getMandatoryEnvVariable(DISABLE_IXBRL_VALIDATION_ENV_VAR))) {
+            //TODO TNEP validation needs to be added. Copy logic from abridged. Next PR
```

Comment:
Can you link this to a story instead please? This TODO will be confusing to other developers if something else becomes priority and you don't implement the validation logic next.